projects/core/src/modules/scrollable-host/scrollable-host.service.ts

```diff
@@ -14,30 +14,109 @@ export class SkyScrollableHostService {
     private windowRef: SkyAppWindowRef
   ) { }
 
-  public getScrollabeHost(elementRef: ElementRef): HTMLElement | Window {
+  /**
+   * Returns the given element's current scrollable host
+   * @param elementRef The element whose scrollable host is being requested
+   * @returns The current scrollable host
+   */
+  public getScrollableHost(elementRef: ElementRef): HTMLElement | Window {
     return this.findScrollableHost(elementRef.nativeElement);
   }
 
-  public watchScrollableHost(elementRef: ElementRef, completionObservable: Observable<void>): Observable<HTMLElement | Window> {
-    let scrollableHost = this.findScrollableHost(elementRef.nativeElement);
-    let behaviorSubject = new BehaviorSubject(scrollableHost);
-
-    const mutationObserver = this.mutationObserverSvc.create(() => {
-      let newScrollableHost = this.findScrollableHost(elementRef.nativeElement);
-
-      if (newScrollableHost !== scrollableHost) {
-        scrollableHost = newScrollableHost;
+  /**
+   * Returns an observable which emits the given element's current scrollable host
+   * @param elementRef The element whose scrollable host is being requested
+   * @param completionObservable An observable which alerts the internal observers that they should complete
+   * @returns An observable which emits the current scrollable host
+   * @internal
+   */
+  public watchScrollableHost(elementRef: ElementRef): Observable<HTMLElement | Window> {
+    let subCount = 0;
+    let subscribers: Subscriber<HTMLElement | Window>[] = [];
+    let mutationObserver: MutationObserver;
+
+    return new Observable((newSubscriber) => {
```

Comment:
The use of `newSubscriber` as a parameter name is a bit confusing given that it's referenced in a closure after it's initially passed into the function. I'd suggest using `subscriber` and then either updating subsequent uses of `subscriber` to be something else or factoring the inner functions into their own functions with their own parameters (the latter would be my preference). You have another bit of code where you loop through subscribers and call `next()` on each one, so you can create a function that does that loop and then can use the `subscriber` variable inside it without worrying about the name clashing. I'd place this function outside the class rather than making it a private method of the class itself since it minifies better and doesn't reference `this`. Note the use of `unknown` rather than `any` here since `unknown` [offers better type safety](https://dmitripavlutin.com/typescript-unknown-vs-any/).

```
function notifySubscribers(subscribers: Subscriber<unknown>, item: unknown) {
  for (const subscriber of subscribers) {
    subscriber.next(item);
  }
}
```